src/java/org/apache/cassandra/index/Index.java

```diff
@@ -426,6 +426,16 @@ public Indexer indexerFor(DecoratedKey key,
                               WriteContext ctx,
                               IndexTransaction.Type transactionType);
 
+    /**
+     * Tells whether this index supports replica fitering protection or not.
+     *
+     * @return true if this index supports replica filtering protection, false otherwise
+     */
+    default boolean supportsReplicaFilteringProtection()
+    {
+        return true;
```

Comment:
Not sure what should be the default argument, I think that for compatibility we should return `false`.

Reply:
@adelapena @maedhroz is not it true that as of today, all indexes are supporting filtering protection but SASI does not? All implementations of a custom index do support filtering based on the current code as it is, only the impls of SASI ones are not so having this on "false" as default does not make sense to me. We would have to go over all impls of indexes and change it to true there or I am missing something. I would make the default what is used more frequently and it seems we are supporting filtering for all indexes but not sasi ones.

Reply:
ehhhhhhh wait, I need to reconsider this.

Reply:
The only index implementation in the codebase other than SASI is regular indexes (`CassandraIndex` and its subclasses). Regular indexes aren't custom indexes so the new method won't ever be called for them. Instead RFP is always used with regular indexes. There are also some mock index implementations in unit tests but they are not used in a way that RFP is involved. So regarding indexes in Cassandra codebase the default value is only relevant for SASI, which is a minimal change. As for third party indexes out there, the question for the default value is not what value is more frequently used but what value is safer in case the implementations miss the changes in the index API and forget to implement the method. It's worth mentioning that previously existing implementations might be already compatible with SRP without any changes if they don't use custom syntax nor transformations of the indexed data. On the one hand, I'd say that indexes with wrongly enabled SRP are far worse than indexes with wrongly disabled SRP, which leads me to prefer a default of `false` for the sake of compatibility. On the other hand, it's very desirable to have SRP enabled since it's a bug fix, and a wrongly enabled SRP is far easy to detect, which makes me prefer a default of `true` for the sake of correctness. wdyt?

Reply:
Thanks for your in-depth answer, I would go for `true` too, hence there is nothing to change here. I am resolving this.